gstd-meta/src/internal.rs

```diff
@@ -0,0 +1,219 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::prelude::{BTreeMap, String, ToString, Vec};
+use scale_info::{IntoPortable, MetaType, Registry};
+use serde_json::to_value;
+
+pub(crate) fn inspect(meta_type: MetaType) -> (String, BTreeMap<String, String>) {
+    let type_info = meta_type.type_info();
+
+    let name = type_info.path().ident();
+    let mut map = BTreeMap::<String, String>::new();
```

Comment:
No need to point to the concrete type when it can be inferred automatically.
```suggestion
    let mut map = BTreeMap::new();
```